### PR TITLE
fix(tree-item): Pressing right arrow key focuses first child when the tree-item contains an expanded subtree

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #3123

## Summary

This is a partial fix for #3123 that restores the correct right arrow key behavior.  The fix now focuses the first child in an expanded subtree.

### CODE REVIEW

File: src/components/tree-item/tree-item.e2e.ts [hunk 1]
```diff
@@ -292,4 +292,59 @@ describe("calcite-tree-item", () => {
...
+              <calcite-tree-item id="instrument">Instrument Cable</calcite-tree-item>
+            </calcite-tree>
+          </calcite-tree-item>
+        </calcite-tree>
+    `
+    });
+
+    await page.keyboard.press("Tab");
+
+    expect(await page.evaluate(() => document.activeElement.id)).toBe("cables");
```
Comment (jcfranco) [inline src/components/tree-item/tree-item.e2e.ts:313]: Nitpick: you could return the values you're asserting on with the same `evaluate` callback.

File: src/components/tree-item/tree-item.tsx [hunk 2]
```diff
@@ -345,7 +345,8 @@ export class TreeItem implements ConditionalSlotComponent {
 
         // When focus is on a open node, moves focus to the first child node.
         if (this.hasChildren && this.expanded) {
-          this.el.querySelector("calcite-tree-item").focus();
+          this.el.querySelector("calcite-tree-item")?.focus();
+          e.stopPropagation();
```
Comment (jcfranco) [inline src/components/tree-item/tree-item.tsx:349]: I see that this is an existing pattern in this implementation, but unless it's an internal event, let's avoid stopping propagation of events. It breaks expectations and could lead to other bugs (e.g., end-user expects to handle key event, but doesn't receive it, so they resort to listening on the capture phase, etc...).

**Review by jcfranco — APPROVED** [review]
Awesome! Just had one comment regarding stopping event propagation.

File: src/components/tree-item/tree-item.tsx (hunk 2 above)
Comment (eriklharper) [reply src/components/tree-item/tree-item.tsx:349]: I'll try and see if there is a better way to solve this in the coming iterations on this bug.

**Pull request merged**